### PR TITLE
handle simulator background refresh error in phone auth

### PR DESCRIPTION
**Pull request by morganchen12**
Fixes a case where Auth would return an error instead of presenting reCAPTCHA when running phone auth in the iOS simulator.

### CODE REVIEW

File: Firebase/Auth/Source/AuthProviders/Phone/FIRPhoneAuthProvider.m [hunk 1]
```diff
@@ -353,7 +353,14 @@ - (void)verifyClientAndSendVerificationCodeToPhoneNumber:(NSString *)phoneNumber
                                                 callback:(FIRVerificationResultCallback)callback {
   [self verifyClientWithCompletion:^(FIRAuthAppCredential *_Nullable appCredential,
                                      NSError *_Nullable error) {
-    if (error) {
+    NSError *_Nullable underlyingError = error.userInfo[NSUnderlyingErrorKey];
+    BOOL isErrorSimulatorError = error.code == FIRAuthErrorCodeMissingAppToken
```
Comment (paulb777) [inline Firebase/Auth/Source/AuthProviders/Phone/FIRPhoneAuthProvider.m]: How about wrapping in ifdef simulator code, so it only gets built for simulator libraries

Comment (morganchen12) [reply Firebase/Auth/Source/AuthProviders/Phone/FIRPhoneAuthProvider.m]: That should make it only available within simulator slices of fat binaries, right?

Comment (paulb777) [reply Firebase/Auth/Source/AuthProviders/Phone/FIRPhoneAuthProvider.m]: Yep, that's another way of saying what I said :)

Comment (morganchen12) [reply Firebase/Auth/Source/AuthProviders/Phone/FIRPhoneAuthProvider.m]: Done.

**Comment by paulb777** [comment]
LGTM, needs approval from @protocol86

**Comment by protocol86** [comment]
when odes this happen? which specific chain of events triggers this particular error?

**Comment by morganchen12** [comment]
Talked with @protocol86 and this PR doesn't address the issue. Closing for now.

**Pull request closed without merging**